### ISSUE TITLE
Add a doc for deprecations and breaking changes

### ISSUE THREAD

**Issue by kaushikb11** [Lightning-AI/lightning#12543]
## Proposal

We should have a dedicated page to track deprecations, breaking changes and optional alternatives in our documentation.
Similar to [this](https://www.sphinx-doc.org/en/master/extdev/deprecated.html).

### Motivation

It should be easier for the users to discover deprecations, upcoming breaking changes and alternatives at a single place.

### Pitch

A doc (deprecations.rst) with a table with APIs, depreciation and removal major versions as well as the alternatives.

```table

| API | Deprecated version | Removal version | Alternatives |
```

<img width="353" alt="Screen Shot 2022-03-31 at 2 05 12 PM" src="https://user-images.githubusercontent.com/45285388/161013327-fc3d5db3-aa3f-494a-9567-3d0ee5da36fa.png">

______________________________________________________________________

#### If you enjoy Lightning, check out our other projects! ⚡

- [**Metrics**](https://github.com/PyTorchLightning/metrics): Machine learning metrics for distributed, scalable PyTorch applications.

- [**Lite**](https://pytorch-lightning.readthedocs.io/en/latest/starter/lightning_lite.html): enables pure PyTorch users to scale their existing code on any kind of device while retaining full control over their own loops and optimization logic.

- [**Flash**](https://github.com/PyTorchLightning/lightning-flash): The fastest way to get a Lightning baseline! A collection of tasks for fast prototyping, baselining, fine-tuning, and solving problems with deep learning.

- [**Bolts**](https://github.com/PyTorchLightning/lightning-bolts): Pretrained SOTA Deep Learning models, callbacks, and more for research and production with PyTorch Lightning and PyTorch.

- [**Lightning Transformers**](https://github.com/PyTorchLightning/lightning-transformers): Flexible interface for high-performance research using SOTA Transformers leveraging Pytorch Lightning, Transformers, and Hydra.

cc @borda @rohitgr7

**Comment by Chandu-4444** [comment]
I'd like to take up this issue. Some pointers would be of great help to me as I'm new to sphinx?

**Comment by carmocca** [comment]
Will be doing this in the release notes. For example, 1.7 will include the section:

# Deprecations

| API                                                                                                          | Removal version | Alternative                                     |
|--------------------------------------------------------------------------------------------------------------|-----------------|-------------------------------------------------|
| Import `pytorch_lightning.loggers.base.LightningLoggerBase`                                                  | 1.9             | `pytorch_lightning.loggers.logger.Logger`       |
| Import `pytorch_lightning.callbacks.base.Callback`                                                           | 1.9             | `pytorch_lightning.callbacks.callback.Callback` |
| Import `pytorch_lightning.core.lightning.LightningModule`                                                    | 1.9             | `pytorch_lightning.core.module.LightningModule` |
| Import `pytorch_lightning.loops.base.Loop`                                                                   | 1.9             | `pytorch_lightning.loops.loop.Loop`             |
| Import `pytorch_lightning.profiler`                                                                          | 1.9             | `pytorch_lightning.profilers`                   |
| Arguments `Trainer(num_processes=..., gpus=..., tpu_cores=..., ipus=...)`                                    | 2.0             | `Trainer(accelerator=..., devices=...)`         |
| Argument `LightningCLI(seed_everything_default=None)`                                                        | 1.9             | `LightningCLI(seed_everything_default=False)`   |
| Method `Trainer.reset_train_val_dataloaders()`                                                               | 1.9             | `Trainer.reset_{train,val}_dataloader`          |
| Import `pytorch_lightning.utilities.cli` module                                                              | 1.9             | `pytorch_lightning.cli`                         |
| Objects `pytorch_lightning.utilities.cli.{OPTIMIZER,LR_SCHEDULER,MODEL,DATAMODULE,CALLBACK,LOGGER}_REGISTRY` | 1.9             | Not necessary anymore                           |
| Argument `LightningCLI(auto_registry=...)`                                                                   | 1.9             | Not necessary anymore                           |
| Argument `Trainer(strategy="ddp2")` and class `pytorch_lightning.strategies.DDP2Strategy`                    | 1.8             | No longer supported                             |

**Comment by awaelchli** [comment]
I believe this is now covered by [the migration guide](https://lightning.ai/docs/pytorch/latest/past_versions.html) implemented in #17034.
